xarray/backends/api.py

```diff
@@ -236,9 +236,16 @@ def _protect_dataset_variables_inplace(dataset, cache):
     for name, variable in dataset.variables.items():
         if name not in dataset._indexes:
             # no need to protect IndexVariable objects
-            data = indexing.CopyOnWriteArray(variable._data)
+
+            if isinstance(variable._data, backends.BackendArray):
```

Comment:
This seems not right to me. We are in `backends/api.py` these should all be BackendArrays.

Reply:
The way that `BackendArray` is presented in [How to add a new backend](https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html#how-to-support-lazy-loading) strongly implies that using `BackendArray` is optional. In the `KerchunkArray` case I specifically don't want to use `BackendArray`, even though I do want to write a backend (from https://github.com/pydata/xarray/pull/8714#discussion_r1480405876):

> my KerchunkArray case is interesting because I don't want to use BackendArray (I have no use for CopyOnWrite because I'm never loading bytes, nor for Lazy indexing (I can't index into the KerchunkArray at all).